src/handlers.rs

```diff
@@ -173,46 +249,27 @@ pub extern "C" fn handle_timer_start(
   debug!("handle_timer_start");
   let deno = from_c(d);
 
-  let when = Instant::now() + Duration::from_millis(delay.into());
   if interval {
-    unimplemented!();
-  }
-
-  let (cancel_tx, cancel_rx) = oneshot::channel::<()>();
+    let (interval_task, cancel_interval) = set_interval(
+      move || {
+        send_timer_ready(d, cmd_id, timer_id, false);
+      },
+      delay,
+    );
 
-  let delay = Delay::new(when)
-    .map_err(|e| panic!("timer failed; err={:?}", e))
-    .and_then(move |_| {
-      let mut builder = flatbuffers::FlatBufferBuilder::new();
-      let msg = msg::CreateTimerReady(
-        &mut builder,
-        &msg::TimerReadyArgs {
-          id: timer_id,
-          done: true,
-          ..Default::default()
-        },
-      );
-      builder.finish(msg);
-      send_base(
-        d,
-        &mut builder,
-        &msg::BaseArgs {
-          cmdId: cmd_id,
-          msg: Some(msg.union()),
-          msg_type: msg::Any::TimerReady,
-          ..Default::default()
-        },
-      );
+    deno.timers.insert(timer_id, cancel_interval);
+    deno.rt.spawn(interval_task);
+  } else {
+    let (delay_task, cancel_delay) = set_timeout(
+      move || {
```

Comment:
Should be here
```rust
    let (delay_task, cancel_delay) = set_timeout(
      move || {
        send_timer_ready(d, cmd_id, timer_id, true);
        let deno = from_c(d);
        deno.timers.remove(&timer_id);
      },
      delay,
    );
```

Reply:
Almost. `remove_timer` needs to be called from the delay task, not the interval task.

Reply:
oops - thanks!